### PR TITLE
Releases are scaled down to 0 after retirement

### PR DESCRIPTION
**Pull request by dokipen**
Hello @ddbenson, @dnelson, @silverlyra, @micahnoland, 

Please review the following commits I made in branch 'dokipen/20190313105719-incarnation-manages-hpa':

Here are some random goodies!

- **Releases are scaled down to 0 after retirement** (963d032d8394464240cf00c510033164b279e589)

- **Move release to targets** (69d2978abebb7e8bee5b9c4820294aaf51178e58)

- **Adds cli flag to turn on route53 management** (fd47c538120cf106cad60742d00440ad627ebd87)

- **Delete orig files after patching crds** (3c2f1a940565f1c50c946aea54dbd3f67da27863)

- **Fixe compilation error** (0958d41e6e4d27c9ec2c4751a4ba0fcbe217242f)

- **Use camelcase in api** (24419b49414a16b98aa2c181b74d31b9bea4d19f)

R=@ddbenson
R=@dnelson
R=@silverlyra
R=@micahnoland

### CODE REVIEW

File: pkg/apis/picchu/v1alpha1/revision_types.go [hunk 1]
```diff
@@ -41,10 +40,6 @@ type RevisionApp struct {
 	Image string `json:"image"`
 }
 
-type RevisionRelease struct {
-	Eligible bool `json:"eligible"`
```
Comment (silverlyra) [inline pkg/apis/picchu/v1alpha1/revision_types.go]: why was this removed?

File: pkg/controller/releasemanager/releasemanager_controller.go [hunk 2]
```diff
@@ -374,6 +374,12 @@ func (r *ResourceSyncer) SyncVirtualService() error {
 			r.Instance.UpdateReleaseStatus(releaseStatus)
 		}
 
+		// Wind down retired releases in HPA
+		// TODO(bob): Use some time-based rule to leave 1 instance running for
```
Comment (dnelson) [inline pkg/controller/releasemanager/releasemanager_controller.go]: I don't think this is necessary. Keeping the release around for a while but turned down to 0 is sufficient.

File: pkg/apis/picchu/v1alpha1/revision_types.go (hunk 1 above)
Comment (dokipen) [reply pkg/apis/picchu/v1alpha1/revision_types.go]: Moved to target.

File: pkg/controller/releasemanager/releasemanager_controller.go (hunk 2 above)
Comment (dokipen) [reply pkg/controller/releasemanager/releasemanager_controller.go]: 👍
